### PR TITLE
[TASK] Upgrade to PHPUnit 8

### LINKED ISSUE #942: Update to PHPUnit 8 (for PHP 7.2)

### PR DESCRIPTION
**Pull request by oliverklee**
Also update the PHPUnit configuration file to use the recommended defaults.

Fixes #942

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -1,12 +1,23 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/7.5/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/8.5/phpunit.xsd"
          bootstrap="vendor/autoload.php"
          colors="true"
+         executionOrder="depends,defects"
```
Comment (JakeQZ) [inline phpunit.xml:6]: I could not find any documentation on what the eight possible values actually mean.

File: phpunit.xml [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
          verbose="true">
     <testsuites>
         <testsuite name="default">
-            <directory suffix="Test.php">tests</directory>
+            <directory>tests</directory>
         </testsuite>
     </testsuites>
+
+    <filter>
+        <whitelist processUncoveredFilesFromWhitelist="true">
```
Comment (JakeQZ) [inline phpunit.xml:19]: Again, struggling to understand what this is doing from the docs.

**Review by JakeQZ — COMMENTED** [review]
I'm finding it hard to have any idea what a couple of the new entries in the XML file actually do.  Are you able to enlighten me?

File: phpunit.xml (hunk 1 above)
Comment (oliverklee) [reply phpunit.xml:6]: depends: If test A depends on test B (as marked by an annotation), execute B before A

defects: Run those tests first that have failed during the previous run

File: phpunit.xml (hunk 2 above)
Comment (oliverklee) [reply phpunit.xml:19]: This means that empty lines and comments can also be marked as executed/covered.

**Review by JakeQZ — APPROVED** [review]
OK.

**Pull request merged**